dev/RadialGradientBrush/TestUI/RadialGradientBrushPage.xaml.cs

```diff
@@ -0,0 +1,254 @@
+
+            DynamicGradientBrush.GradientStops.Clear();
+            for (int i = 0; i < _random.Next(2, 5); i++)
+            {
+                AddRandomGradientStop(DynamicGradientBrush);
+            }
+
+            // Set brush after adding stops
+            TextBlockWithDynamicGradient.Foreground = DynamicGradientBrush;
+
+            Bindings.Update();
+        }
+
+        private void AddGradientStopButton_Click(object sender, RoutedEventArgs e)
+        {
+            AddRandomGradientStop(DynamicGradientBrush);
+        }
+
+        private void RemoveGradientStopButton_Click(object sender, RoutedEventArgs e)
+        {
+            RemoveRandomGradientStop(DynamicGradientBrush);
+        }
+
+        private void RandomizeGradientOriginButton_Click(object sender, RoutedEventArgs e)
+        {
+            RandomizeGradientOriginOffset(DynamicGradientBrush);
+        }
+
+        private void RandomizeEllipseCenterButton_Click(object sender, RoutedEventArgs e)
+        {
+            RandomizeEllipseCenter(DynamicGradientBrush);
+        }
+
+        private void RandomizeEllipseRadiusButton_Click(object sender, RoutedEventArgs e)
+        {
+            RandomizeEllipseRadius(DynamicGradientBrush);
+        }
+
+        private void ToggleMappingModeButton_Click(object sender, RoutedEventArgs e)
+        {
+            ToggleMappingMode(DynamicGradientBrush);
+        }
+
+        private async void GenerateRenderTargetBitmapButton_Click(object sender, RoutedEventArgs e)
+        {
+            var rtb = new RenderTargetBitmap();
+            await rtb.RenderAsync(GradientRectangle);
+
+            RenderTargetBitmapResultRectangle.Fill = new ImageBrush() { ImageSource = rtb };
+
+            var pixelBuffer = await rtb.GetPixelsAsync();
+            byte[] pixelArray = pixelBuffer.ToArray();
+
+            // Sample top left and center pixels to verify rendering is correct.
+            var centerColor = GetPixelAtPoint(new Point(50, 50), rtb, pixelArray);
+            var outerColor = GetPixelAtPoint(new Point(0, 0), rtb, pixelArray);
+
+            if (ApiInformation.IsTypePresent("Windows.UI.Composition.CompositionRadialGradientBrush"))
+            {
+                // If CompositionRadialGradientBrush is available then should be rendering a gradient.
+                if (centerColor == Colors.Orange && outerColor == Colors.Green)
+                {
+                    ColorMatchTestResult.Text = "Passed";
+                }
+                else
+                {
+                    ColorMatchTestResult.Text = "Failed";
+                }
+            }
+            else
+            {
+                if (centerColor == Colors.Red && outerColor == Colors.Red)
+                {
+                    ColorMatchTestResult.Text = "Passed";
+                }
+                else
+                {
+                    ColorMatchTestResult.Text = "Failed";
+                }
+            }
```

Comment:
Interesting, why is this code here rather than in the test? I think generally we would output the center and outer color value to something the test can read rather than outputing a passed or failed state.  As written we would not get very much info from a log of a test failure.

Reply:
I was following the precedent of the acrylic tests which do something similar:

https://github.com/microsoft/microsoft-ui-xaml/blob/1f48f11b3cd895b10206f9abc93defb8eeff26b5/dev/Materials/Acrylic/TestUI/AcrylicRenderingPage.xaml.cs#L137

Reply:
hmm, @ranjeshj @kmahone thoughts? I think having the test page report info to the test is better, that way if the test fails we will see something like 0xFF0001 != 0xFF0000 instead of Failed != Passed